### PR TITLE
[PPP-5353]-XSS Findings For Pentaho-platform-plugin-reporting

### PR DESCRIPTION
**Pull request by Rangashivani**
[PPP-5353]-XSS Findings For Pentaho-platform-plugin-reporting

[PPP-5353]: https://hv-eng.atlassian.net/browse/PPP-5353?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 4m 50s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **540**, Failures: **0**, Skipped: **3** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-reporting-plugin/job/PR-864/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/javascript/reportviewer/dojo/pentaho/reportviewer/GlassPane.js [hunk 1]
```diff
@@ -15,26 +15,26 @@
 *
 */
 define(["dojo/_base/declare", "dijit/_WidgetBase", "dijit/_Templated", "dojo/on", "dojo/query",
-"pentaho/common/button", "pentaho/common/Dialog", "dojo/text!pentaho/reportviewer/GlassPane.html"],
-    function(declare, _WidgetBase, _Templated, on, query, button, Dialog, templateStr){
+"pentaho/common/button", "pentaho/common/Dialog", "dojo/text!pentaho/reportviewer/GlassPane.html","common-ui/util/xss"],
```
Comment (dcleao) [inline core/src/main/javascript/reportviewer/dojo/pentaho/reportviewer/GlassPane.js:18]: Missing space after comma.

File: core/src/main/javascript/reportviewer/reportviewer.js [hunk 2]
```diff
@@ -1193,7 +1193,7 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
                     hideDlgAndPane(registry.byId('feedbackScreen'));
 
                     //Show loading screen
-                    $('#notification-message').html(_Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages);
+                    xssUtil.setHtml($('notification-message'), _Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages);
```
Comment (dcleao) [inline core/src/main/javascript/reportviewer/reportviewer.js:1196]: I can't find the comment from another PR review where I wrote this, so I'm repeating here.

Note that the query selector passed to jQuery should remain as `#notification-message`.

The XSS Utility `setHtml(elem, html)` method does not support jQuery objects in the first argument, only actual `HTMLElement`.

On the other hand, if we keep using the pattern `$('#notification-message').html(xssUtil.sanitizeHtml(html))` with jQuery instances, likely, the corresponding AppScan findings will not be automatically resolved.

As such, we should consider adding support for jQuery objects to the existing `setHtml(..)` method, and then use it with these jQuery cases.

File: core/src/main/javascript/reportviewer/reportviewer.js [hunk 3]
```diff
@@ -1236,7 +1236,7 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
                       isPageCountUpdated = true;
                     }
 
-                    $('#notification-message').html(_Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages);
+                    xssUtil.setHtml($('#notification-message'), _Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages);
```
Comment (dcleao) [inline core/src/main/javascript/reportviewer/reportviewer.js:1239]: Idem, regarding adding jQuery support to `setHtml(..)`.

File: core/src/main/javascript/reportviewer/reportviewer.js [hunk 4]
```diff
@@ -1246,13 +1246,13 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
                 case "QUEUED":
                 case "WORKING":
                   // Although we are hiding the screen latter, update the label anyway;
-                  $('#notification-message').html(_Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages);
+                  xssUtil.setHtml($('#notification-message'), _Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages);
```
Comment (dcleao) [inline core/src/main/javascript/reportviewer/reportviewer.js:1249]: Idem, regarding adding jQuery support to `setHtml(..)`.

File: core/src/main/javascript/reportviewer/reportviewer.js [hunk 5]
```diff
@@ -1246,13 +1246,13 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
...
                   me._hideAsyncScreens();
                   me._keepPolling(mainJobStatus.uuid, url, mainReportGeneration);
                   break;
                 case "FINISHED":
                   // Although we are hiding the screen latter, update the label anyway;
-                  $('#notification-message').html(_Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages);
+                  xssUtil.setHtml($('#notification-message'), _Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages);
```
Comment (dcleao) [inline core/src/main/javascript/reportviewer/reportviewer.js:1255]: Idem, regarding adding jQuery support to `setHtml(..)`.

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 58s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **540**, Failures: **0**, Skipped: **3** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-reporting-plugin/job/PR-864/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request closed without merging**